frontend/src/modules/Metadata_Forms/components/MetadataFormEnforcement.js

```diff
@@ -1,21 +1,781 @@
+import { listOrganizations } from '../../Organizations/services';
+import { listDatasets } from '../../DatasetsBase/services';
+import { useTheme } from '@mui/styles';
+import DeleteIcon from '@mui/icons-material/DeleteOutlined';
+import { DataGrid } from '@mui/x-data-grid';
+
+const CreateEnforcementRuleModal = (props) => {
+  const {
+    onCancel,
+    onSubmit,
+    open,
+    metadataForm,
+    severityOptions,
+    entityTypesOptions,
+    enforcementScopeOptions,
+    ...other
+  } = props;
+
+  const client = useClient();
+  const dispatch = useDispatch();
+
+  const [environmentOptions, setEnvironmentOptions] = useState([]);
+  const [organizationOptions, setOrganizationOptions] = useState([]);
+  const [datasetOptions, setDatasetOptions] = useState([]);
+  const [entityTypes, setEntityTypes] = useState([...entityTypesOptions]);
+
+  const enforcementScopeDict = {};
+  for (const option of enforcementScopeOptions) {
+    enforcementScopeDict[option.name] = option.value;
+  }
+
+  const fetchOrganizations = async () => {
+    try {
+      const response = await client.query(
+        listOrganizations({
+          filter: Defaults.selectListFilter
+        })
+      );
+      if (!response.errors) {
+        setOrganizationOptions(
+          response.data.listOrganizations.nodes.map((e) => ({
+            ...e,
+            value: e.organizationUri,
+            label: e.label
+          }))
+        );
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
+    } catch (e) {
+      dispatch({ type: SET_ERROR, error: e.message });
+    }
+  };
+  const fetchEnvironments = async () => {
+    try {
+      const response = await client.query(
+        listValidEnvironments({
+          filter: Defaults.selectListFilter
+        })
+      );
+      if (!response.errors) {
+        setEnvironmentOptions(
+          response.data.listValidEnvironments.nodes.map((e) => ({
+            ...e,
+            value: e.environmentUri,
+            label: e.label
+          }))
+        );
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
+    } catch (e) {
+      dispatch({ type: SET_ERROR, error: e.message });
+    }
+  };
+
+  const fetchDatasets = async () => {
+    try {
+      const response = await client.query(
+        listDatasets({
+          filter: Defaults.selectListFilter
+        })
+      );
+      if (!response.errors) {
+        setDatasetOptions(
+          response.data.listDatasets.nodes.map((e) => ({
+            ...e,
+            value: e.datasetUri,
+            label: e.label
+          }))
+        );
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
+    } catch (e) {
+      dispatch({ type: SET_ERROR, error: e.message });
+    }
+  };
+
+  useEffect(async () => {
+    if (client) {
+      await fetchEnvironments();
+      await fetchOrganizations();
+      await fetchDatasets();
```

Comment:
I am not sure of this, but can we restrict the fetchX call to call only the one of the enforcementScope selected? If we are using the Organizations, we do not need to fetch Environments and Datasets and we can save 2 costly API calls